pandas/core/arrays/datetimes.py

```diff
@@ -207,26 +215,46 @@ def __new__(cls, values, freq=None, tz=None, dtype=None):
         # if dtype has an embedded tz, capture it
         tz = dtl.validate_tz_from_dtype(dtype, tz)
 
-        if isinstance(values, DatetimeArrayMixin):
+        if lib.is_scalar(values):
+            raise TypeError(dtl.scalar_data_error(values, cls))
+        elif isinstance(values, ABCSeries):
+            # extract nanosecond unix timestamps
+            if tz is None:
+                # TODO: Try to do this in just one place
+                tz = values.dt.tz
+            values = np.array(values.view('i8'))
+        elif isinstance(values, DatetimeArrayMixin):
```

Comment:
> DatetimeArrayMixin -> cls?

No.  For the moment we are still using inheritance, so this would mess up for DatetimeIndex == DatetimeArray.  When we change to composition this check will have to become `isinstance(values, (DatetimeArray, ABCDatetimeIndex))`